From c0d91fb6c332e9d284b23f8b647ede836c48698b Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Mon, 3 May 2021 16:20:21 +1000
Subject: [PATCH] Add RoboStack OS detection (#1)

---
 src/rospkg/os_detect.py | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/rospkg/os_detect.py b/src/rospkg/os_detect.py
index d1ead71..509dfed 100644
--- a/src/rospkg/os_detect.py
+++ b/src/rospkg/os_detect.py
@@ -242,6 +242,24 @@ def get_codename(self):
         raise OsNotDetected('called in incorrect OS')
 
 
+class RoboStack(OsDetector):
+    """
+    Detect RoboStack.
+    """
+    def is_os(self):
+        return "ROS_OS_OVERRIDE" in os.environ and os.environ["ROS_OS_OVERRIDE"].startswith("robostack")
+
+    def get_version(self):
+        if self.is_os():
+            return ""
+        raise OsNotDetected('called in incorrect OS')
+
+    def get_codename(self):
+        if self.is_os():
+            return ""
+        raise OsNotDetected('called in incorrect OS')
+
+
 class OpenSuse(OsDetector):
     """
     Detect OpenSuse OS.
@@ -645,6 +663,7 @@ def get_codename(self):
 OS_OPENSUSE = 'opensuse'
 OS_OPENSUSE13 = 'opensuse'
 OS_ORACLE = 'oracle'
+OS_ROBOSTACK = 'robostack'
 OS_TIZEN = 'tizen'
 OS_SAILFISHOS = 'sailfishos'
 OS_OSX = 'osx'
@@ -686,6 +705,7 @@ def get_codename(self):
 OsDetect.register_default(OS_OPENSUSE, FdoDetect("opensuse-leap"))
 OsDetect.register_default(OS_OPENSUSE, FdoDetect("opensuse"))
 OsDetect.register_default(OS_ORACLE, FdoDetect("ol"))
+OsDetect.register_default(OS_ROBOSTACK, RoboStack())
 OsDetect.register_default(OS_TIZEN, FdoDetect("tizen"))
 OsDetect.register_default(OS_SAILFISHOS, FdoDetect("sailfishos"))
 OsDetect.register_default(OS_OSX, OSX())
